image-extension.md

```diff
@@ -0,0 +1,201 @@
+# Image Extension Interface Specification
+
+This document specifies the interface of image extensions.
+
+Image extensions are dynamically-generated or static build-time and runtime modules that mutate the pre-build base image.
+
+## Table of Contents
+
+<!-- Using https://github.com/yzhang-gh/vscode-markdown to manage toc -->
+
+
+## Image Extension API Version
+
+This document specifies Image Extension API version `0.1`.
+
+Image Extension API versions:
+ - MUST be in form `<major>.<minor>` or `<major>`, where `<major>` is equivalent to `<major>.0`
+ - When `<major>` is greater than `0` increments to `<minor>` SHALL exclusively indicate additive changes
+
+## Image Extension Interface
+
+An image extension is used to apply a `Dockerfile` at the platform's discretion.
+
+The image extensions MAY contain a `/bin/build` and `/bin/detect` executable.
+Each extensions MUST have an `extension.toml` file in its root directory.
+
+Extensions participate in the buildpack [detection](#detector) process, with the same `UID`, `GID`, and interface for `//bin/detect`. However:
+- Detection is optional for extensions, and they are assumed to pass detection when it is not present. A `/bin/detect` that exits with a 0 exit code passes detection, and fails otherwise.
```

Comment:
```suggestion
- Detection is optional for extensions, and they are assumed to pass detection when `/bin/detect` is not present. A `/bin/detect` that exits with a 0 exit code passes detection, and fails otherwise.
```